src/components/overview/teamIntervalsOverview.tsx

```diff
@@ -0,0 +1,178 @@
+  ResponsiveContainer,
+  Line,
+  LineChart,
+  AreaChart,
+  Area,
+} from "recharts";
+import {
+  ValueType,
+  NameType,
+} from "recharts/types/component/DefaultTooltipContent";
+import * as React from "react";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { TeamIntervalsOverviewDto } from "@/app/api/teams/overview/intervals/route";
+
+const CustomTooltip = ({
+  active,
+  payload,
+}: TooltipProps<ValueType, NameType>) => {
+  if (active && payload && payload.length) {
+    return (
+      <div className="custom-tooltip">
+        {payload.map((data) => {
+          return (
+            <div>
+              <p className="label">{`${data.dataKey}: ${parseFloat(data.value!.toString()).toPrecision(3)}`}</p>
+            </div>
+          );
+        })}
+      </div>
+    );
+  }
+
+  return null;
+};
+
+interface TeamIntervalsOverviewProps {
+  teamIntervalsOverview: TeamIntervalsOverviewDto[];
+}
+
+interface TeamIntervalsChartData {
+  start: number;
+  end: number;
+  teamScores: { name: string; score: number }[];
+}
+
+function indexToColor(index: number): string {
+  //switch case for index
+  switch (index) {
+    case 0:
+      return "#adfa1d";
+    case 1:
+      return "#A4FFFF";
+    case 2:
+      return "#FFB86C";
+    case 3:
+      return "#242c54";
+    case 4:
+      return "#A4FFFF";
+    case 5:
+      return "#FFFFFF";
+    default:
+      return "#FFFFFF";
+  }
+}
+
+export function TeamIntervalsOverview({
+  teamIntervalsOverview,
+}: TeamIntervalsOverviewProps) {
+  const uniqueTeamNames = teamIntervalsOverview.map((team) => {
+    return team.teamName;
+  });
+  const teamNameToIndex = new Map<string, number>();
+  uniqueTeamNames.forEach((team, index) => {
+    teamNameToIndex.set(team, index);
+  });
+  let teamIntervalChartData: TeamIntervalsChartData[] = [];
+  const scoresMappedByStart = new Map<
+    number,
+    { name: string; score: number }[]
+  >();
+  const endMappedByStart = new Map<number, number>();
+  const uniqueStarts: number[] = [];
+  let counter = 0;
+  for (const team of teamIntervalsOverview) {
+    team.intervals.forEach((interval) => {
+      if (!uniqueStarts.includes(interval.start)) {
+        uniqueStarts.push(interval.start);
+
+        scoresMappedByStart.set(interval.start, [
+          { name: team.teamName, score: interval.score },
+        ]);
+        endMappedByStart.set(interval.start, interval.end);
+      } else {
+        counter += 1;
+        scoresMappedByStart
+          .get(interval.start)
+          ?.push({ name: team.teamName, score: interval.score });
+      }
+    });
+  }
+
+  scoresMappedByStart.forEach((value, key) => {
+    teamIntervalChartData.push({
+      start: key,
+      end: endMappedByStart.get(key) as number,
+      teamScores: value,
+    });
+  });
+
+  teamIntervalChartData.sort((a, b) => a.start - b.start);
+  const currentDate = Date.now();
+  const threeMonthsAgo = currentDate - 7776000000;
```

Comment:
Good idea to move to a const class for this type of variables

Reply:
Actually, maybe we should do this on the backend? aka api level.  Think we have a bit to much business logic on the UI

Reply:
We can't really do that, because we define dynamic key types for the objects, as the data needs to look like:

{ 
start: number
end: number
myFirstTeamName: number
mySecondTeamName: number
...
}

Therefore we can't really expect this data to have some specific type on the frontend.

In the end, even if we use "any" for the api object, when we receive it we will need to type it anyway on the client/component level to use on the chart, so we end up even increasing the complexity of the process, as we will need to transform this data first using the same steps on the backend, and then type and transform it again on the frontend for the chart to be able to use it

Reply:
I miss python with pandas and dataframes...

Reply:
But maybe you meant only the filtering and sorting?

If that's the case I'm adding that on the backend now on the latest commit ✅